OSX test fail - test_ltisys.TestPlacePoles.test_real

I'm experiencing a test failure on current master - OSX 10.12.5.

### Reproducing code example:
```
python -c "import scipy as s; s.test()"
```

### Error message:
```
======================================================================
ERROR: test_ltisys.TestPlacePoles.test_real
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/anz/miniconda3/envs/dev3/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/anz/miniconda3/envs/dev3/lib/python3.6/site-packages/scipy/signal/tests/test_ltisys.py", line 69, in test_real
    self._check(A, B, (2,2,3,3))
  File "/Users/anz/miniconda3/envs/dev3/lib/python3.6/site-packages/scipy/signal/tests/test_ltisys.py", line 45, in _check
    fsf = place_poles(A, B, P, **kwargs)
  File "/Users/anz/miniconda3/envs/dev3/lib/python3.6/site-packages/scipy/signal/ltisys.py", line 3006, in place_poles
    poles, B, maxiter, rtol)
  File "/Users/anz/miniconda3/envs/dev3/lib/python3.6/site-packages/scipy/signal/ltisys.py", line 2646, in _YT_loop
    det_transfer_matrixb = np.abs(np.linalg.det(transfer_matrix))
  File "/Users/anz/miniconda3/envs/dev3/lib/python3.6/site-packages/numpy/linalg/linalg.py", line 1804, in det
    r = _umath_linalg.det(a, signature=signature)
RuntimeWarning: invalid value encountered in det

----------------------------------------------------------------------
```

### Scipy/Numpy/Python version information:
```
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.0.0.dev0+c019b87 1.13.1 sys.version_info(major=3, minor=6, micro=1, releaselevel='final', serial=0)
```

I can still see the issue on my work computer. This morning I rebased on the tippy top of master, then did a "git clean -xdf", followed by "python runtests.py". Unfortunately I don't have time to fix this myself (vacation next week).

```
OSX 10.12.6,
Python 3.6.1 |Continuum Analytics, Inc.| (default, Mar 22 2017, 19:25:17) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
pytest-3.2.1, py-1.4.34, pluggy-0.4.0.
numpy.version.version='1.13.1'
```
The fail this morning is:
```
=================================================== FAILURES ====================================================
___________________________________________ TestPlacePoles.test_real ____________________________________________


self = <scipy.signal.tests.test_ltisys.TestPlacePoles object at 0x1243729e8>

    def test_real(self):
        # Test real pole placement using KNV and YT0 algorithm and example 1 in
        # section 4 of the reference publication (see place_poles docstring)
        A = np.array([1.380, -0.2077, 6.715, -5.676, -0.5814, -4.290, 0,
                      0.6750, 1.067, 4.273, -6.654, 5.893, 0.0480, 4.273,
                      1.343, -2.104]).reshape(4, 4)
        B = np.array([0, 5.679, 1.136, 1.136, 0, 0, -3.146,0]).reshape(4, 2)
        P = np.array([-0.2, -0.5, -5.0566, -8.6659])
    
        # Check that both KNV and YT compute correct K matrix
        self._check(A, B, P, method='KNV0')
        self._check(A, B, P, method='YT')
    
        # Try to reach the specific case in _YT_real where two singular
        # values are almost equal. This is to improve code coverage but I
        # have no way to be sure this code is really reached
    
>       self._check(A, B, (2,2,3,3))

A          = array([[ 1.38  , -0.2077,  6.715 , -5.676 ],
       [-0.5814, -4.29  ,  0.    ,  0.675 ],
       [ 1.067 ,  4.273 , -6.654 ,  5.893 ],
       [ 0.048 ,  4.273 ,  1.343 , -2.104 ]])
B          = array([[ 0.   ,  5.679],
       [ 1.136,  1.136],
       [ 0.   ,  0.   ],
       [-3.146,  0.   ]])
P          = array([-0.2   , -0.5   , -5.0566, -8.6659])
self       = <scipy.signal.tests.test_ltisys.TestPlacePoles object at 0x1243729e8>

scipy/signal/tests/test_ltisys.py:71: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
scipy/signal/tests/test_ltisys.py:47: in _check
    fsf = place_poles(A, B, P, **kwargs)
scipy/signal/ltisys.py:3006: in place_poles
    poles, B, maxiter, rtol)
scipy/signal/ltisys.py:2646: in _YT_loop
    det_transfer_matrixb = np.abs(np.linalg.det(transfer_matrix))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = array([[ 0.19217467,  0.19217467,  0.24827647,  0.24827647],
       [-0.06100367, -0.06100367, -0.08944558, -0.0894455...    [ 0.54689237,  0.54689237,  0.49367072,  0.49367072],
       [ 0.81256149,  0.81256149,  0.82864196,  0.82864196]])

    def det(a):
        """
        Compute the determinant of an array.
    
        Parameters
        ----------
        a : (..., M, M) array_like
            Input array to compute determinants for.
    
        Returns
        -------
        det : (...) array_like
            Determinant of `a`.
    
        See Also
        --------
        slogdet : Another way to representing the determinant, more suitable
          for large matrices where underflow/overflow may occur.
    
        Notes
        -----
    
        .. versionadded:: 1.8.0
    
        Broadcasting rules apply, see the `numpy.linalg` documentation for
        details.
    
        The determinant is computed via LU factorization using the LAPACK
        routine z/dgetrf.
    
        Examples
        --------
        The determinant of a 2-D array [[a, b], [c, d]] is ad - bc:
    
        >>> a = np.array([[1, 2], [3, 4]])
        >>> np.linalg.det(a)
        -2.0
    
        Computing determinants for a stack of matrices:
    
        >>> a = np.array([ [[1, 2], [3, 4]], [[1, 2], [2, 1]], [[1, 3], [3, 1]] ])
        >>> a.shape
        (3, 2, 2)
        >>> np.linalg.det(a)
        array([-2., -3., -8.])
    
        """
        a = asarray(a)
        _assertRankAtLeast2(a)
        _assertNdSquareness(a)
        t, result_t = _commonType(a)
        signature = 'D->D' if isComplexType(t) else 'd->d'
>       r = _umath_linalg.det(a, signature=signature)
E       RuntimeWarning: invalid value encountered in det

a          = array([[ 0.19217467,  0.19217467,  0.24827647,  0.24827647],
       [-0.06100367, -0.06100367, -0.08944558, -0.0894455...    [ 0.54689237,  0.54689237,  0.49367072,  0.49367072],
       [ 0.81256149,  0.81256149,  0.82864196,  0.82864196]])
result_t   = <class 'numpy.float64'>
signature  = 'd->d'
t          = <class 'numpy.float64'>

/Users/anz/miniconda3/envs/dev3/lib/python3.6/site-packages/numpy/linalg/linalg.py:1804: RuntimeWarning
============================================= 577 tests deselected ==============================================
========= 1 failed, 12086 passed, 1347 skipped, 577 deselected, 73 xfailed, 4 xpassed in 446.08 seconds =========
 DVODE--  At current T (=R1), MXSTEP (=I1) steps   
       taken on this call before reaching TOUT     
      In above message,  I1 =         1
      In above message,  R1 =  0.2569611864708D-12
 ZVODE--  At current T (=R1), MXSTEP (=I1) steps   
       taken on this call before reaching TOUT     
      In above message,  I1 =         1
      In above message,  R1 =  0.2569611864708D-12
 lsoda--  at current t (=r1), mxstep (=i1) steps   
       taken on this call before reaching tout     
      in above message,  i1 =         1
      in above message,  r1 =  0.9999999999999D-09
```

--- a/scipy/signal/tests/test_ltisys.py
+++ b/scipy/signal/tests/test_ltisys.py
@@ -68,7 +68,10 @@ class TestPlacePoles(object):
         # values are almost equal. This is to improve code coverage but I
         # have no way to be sure this code is really reached
 
-        self._check(A, B, (2,2,3,3))
+        # on some architectures this can lead to a RuntimeWarning invalid
+        # value in divide (see gh-7590), so suppress it for now
+        with np.errstate(invalid='ignore'):
+            self._check(A, B, (2,2,3,3))
 
     def test_complex(self):
         # Test complex pole placement on a linearized car model, taken from L.

